dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -689,6 +714,35 @@ private Optional<? extends Version<?>> getVersion(Entry<?, ?> entry, String vers
         return versionOptional;
     }
 
+    /**
+     * Get candidate version that can be considered for an ai Topic (or other?)
+     * @param entry
+     * @return
+     */
+    private Optional<? extends Version<?>> getVersionCandidate(Entry<?, ?> entry) {
+        if (entry instanceof Workflow workflow) {
+            List<WorkflowVersion> workflowVersions = workflowVersionDAO.getWorkflowVersionsByWorkflowId(workflow.getId(), 1, 1);
+            if (!workflowVersions.isEmpty()) {
+                return Optional.of(workflowVersions.get(0));
```

Comment:
Shouldn't we be using the default version, if there is one? And if not, should it be the most recently updated version?

Reply:
This actually calls https://github.com/dockstore/dockstore/blob/2f81db3457d3227bcc03449bc80ad490cfba1ef8/dockstore-webservice/src/main/java/io/dockstore/webservice/core/WorkflowVersion.java#L61 which means that it is the most recently updated version as per the test. 

There's an argument to be made for using the default version especially if `latestTagAsDefault` is turned on. 
https://docs.dockstore.org/en/stable/assets/templates/workflows/workflows.html

This would probably reduce the number of ai topics we need to recompute by a lot, on the other hand the default version *is* the version that other users should be seeing if the default version is being set properly, which I have a bit of doubt about.